test/literal_gen.py

```diff
@@ -0,0 +1,49 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+import random
+from string import digits, hexdigits
+
+def joiner(*items):
+    return ''.join(item() for item in items)  
+
+def roll(item, n1, n2=None):
+    n2 = n2 or n1
+    return lambda: ''.join(item() for _ in xrange(random.randint(n1, n2)))
```

Comment:
It's a very bad idea to use random values in tests - you'll chase the ghosts here. Try [hypothesis](https://github.com/DRMacIver/hypothesis) instead.